### PR TITLE
Add tokio-console.md to spec documentation

### PR DESCRIPTION
**Pull request by Adam-Gleave**
# Description of change

Add `console.md` to `/docs/dev/specs`, describing the use of the `console` feature.

## Type of change
- Documentation Fix

### CODE REVIEW

**Comment by thibault-martinez** [comment]
Can you rename it to `tokio-console.md` please ?

**Comment by Adam-Gleave** [comment]
Done 👍

**Review by thibault-martinez — COMMENTED** [review]
I would probably add a little bit of context. Why is this a nice feature ? What can we do with it that we can't do without ? etc

**Comment by Adam-Gleave** [comment]
Should have more context into why we want this feature now and what it can do for us. Let me know if there's anything else specific that you want to see.

**Review by StromFLIX — DISMISSED** [review]

**Review by johanneskeim — DISMISSED** [review]

File: docs/dev/specs/tokio-console.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# `console` feature
+The `bee-node` crate now includes a `console` feature, which can help with debugging and providing diagnostics for the multiple async tasks that Bee is executing at any given time. Building with this feature enabled and running a separate executable will provide realtime diagnostics in your terminal. 
+
+# What is it?
+The `console` feature provides instrumentation on all tasks spawned in the `BeeNode::spawn` method, and aggregates information provided by this insrumentation in a "subscriber", that allows logging and further processing. This is extremely useful for Bee particularly, since it makes use of a task-heavy architecture that can be difficult to debug and diagnose: having real-time task diagnostics should help to alleviate this.
```
Comment (kilianhln) [inline docs/dev/specs/tokio-console.md]: Typo insrumentation -> instrumentation

Comment (Adam-Gleave) [reply docs/dev/specs/tokio-console.md]: Fixed, thanks!

**Review by pvdrz — DISMISSED** [review]

File: docs/dev/specs/tokio-console.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
+# `console` feature
+The `bee-node` crate now includes a `console` feature, which can help with debugging and providing diagnostics for the multiple async tasks that Bee is executing at any given time. Building with this feature enabled and running a separate executable will provide realtime diagnostics in your terminal.
```
Comment (thibault-martinez) [inline docs/dev/specs/tokio-console.md]:
```suggestion
The `bee-node` crate includes a `console` feature, which can help with debugging and providing diagnostics for the multiple async tasks that Bee is executing at any given time. Building with this feature enabled and running a separate executable will provide realtime diagnostics in your terminal. 
```

Comment (thibault-martinez) [inline docs/dev/specs/tokio-console.md]:
```suggestion
The `bee-node` crate now includes a `console` feature, which can help with debugging and providing diagnostics for the multiple async tasks that the node is executing at any given time. Building with this feature enabled and running a separate executable will provide realtime diagnostics in your terminal. 
```

File: docs/dev/specs/tokio-console.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
+# `console` feature
+The `bee-node` crate now includes a `console` feature, which can help with debugging and providing diagnostics for the multiple async tasks that Bee is executing at any given time. Building with this feature enabled and running a separate executable will provide realtime diagnostics in your terminal. 
+
+# What is it?
+The `console` feature provides instrumentation on all tasks spawned in the `BeeNode::spawn` method, and aggregates information provided by this instrumentation in a "subscriber", that allows logging and further processing. This is extremely useful for Bee particularly, since it makes use of a task-heavy architecture that can be difficult to debug and diagnose: having real-time task diagnostics should help to alleviate this.
```
Comment (thibault-martinez) [inline docs/dev/specs/tokio-console.md]: Only tasks spawned in `BeeNode::spawn` ?

Comment (thibault-martinez) [inline docs/dev/specs/tokio-console.md]: You use both `realtime` and `real-time`, align!

File: docs/dev/specs/tokio-console.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+# How do I use it?
+To build/run with the instrumentation, make sure the `console` feature is enabled:
+```
+> cargo run --release --features console
+```
+Additionally, your `RUSTFLAGS` environment variable must contain the following, in order for `tokio` to be built with the `tracing` feature enabled:
+```
+--cfg tokio_unstable
+```
```
Comment (thibault-martinez) [inline docs/dev/specs/tokio-console.md:37]: I would maybe combine the two previous sentences and the following command `RUSTFLAGS="--cfg tokio_unstable" cargo run --release --features console`

File: docs/dev/specs/tokio-console.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+Once the node is up and running, clone a local copy of the [`tokio-console`](https://github.com/tokio-rs/console) project, and run the `console` crate from its directory:
+```
+.../tokio-console/console/ > cargo run
```
Comment (thibault-martinez) [inline docs/dev/specs/tokio-console.md]: For some reason it took me a while to understand that the `>` is not a redirection. Maybe use the standard `$` to indicate start of the command.

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: docs/dev/specs/tokio-console.md (hunk 3 above)
Comment (Adam-Gleave) [reply docs/dev/specs/tokio-console.md]: Yep, there's a little boilerplate so currently it's all contained here. This way all worker tasks can be instrumented from one place. There are a couple extra tasks spawned in the node, but all look to be nested within these worker tasks -- nested tasks can also trigger the deadlock we're seeing.

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**